releasenotes/notes/paulifeaturemap-takes-dictionary-as-entanglement-02037cb2d46e1c41.yaml

```diff
@@ -0,0 +1,20 @@
+---
+features_circuits:
+  - |
+    The entanglement argument in :class:`.PauliFeatureMap` and :class:`.ZZFeatureMap` will no longer support ``List[int]`` or ``List[List[int]]``. Instead,
+    the entanglement structure in :class:`.PauliFeatureMap` and :class:`.ZZFeatureMap` can now be specified as a dictionary where the keys represent  
+    the number of qubits, and the values are lists of integer tuples that define which qubits are entangled with one another. This
+    allows for more flexibility in constructing feature maps tailored to specific quantum algorithms. 
+    Example usage::
+
+      from qiskit.circuit.library import PauliFeatureMap
+      entanglement = {
+        1: [(0,), (2,)],
+        2: [(0, 1), (1, 2)],
+        3: [(0, 1, 2)],
+      }
+      qc = PauliFeatureMap(3, reps=2, paulis=['Z', 'ZZ', 'ZZZ'], entanglement=entanglement, insert_barriers=True)
+      qc.decompose().draw('mpl')
+
```

Comment:
It was a bug that we could specify the entanglement as lists to begin with -- that didn't make sense 🙂 I reformulated the reno a bit to reflect this:
```suggestion
fixes:
  - |
    Fixed that the entanglement in :class:`.PauliFeatureMap` and :class:`.ZZFeatureMap`
    could be given as ``List[int]`` or ``List[List[int]]``, which was incompatible with the fact
    that entanglement blocks of different sizes are used. Instead, the entanglement can be 
    given as dictionary with ``{block_size: entanglement}`` pairs.
features_circuits:
  - |
    :class:`.PauliFeatureMap` and :class:`.ZZFeatureMap` now support specifying the 
    entanglement as a dictionary where the keys represent the number of qubits, and 
    the values are lists of integer tuples that define which qubits are entangled with one another. This
    allows for more flexibility in constructing feature maps tailored to specific quantum algorithms. 
    Example usage::

      from qiskit.circuit.library import PauliFeatureMap
      entanglement = {
        1: [(0,), (2,)],
        2: [(0, 1), (1, 2)],
        3: [(0, 1, 2)],
      }
      qc = PauliFeatureMap(3, reps=2, paulis=['Z', 'ZZ', 'ZZZ'], entanglement=entanglement, insert_barriers=True)
      qc.decompose().draw('mpl')

```